ecleankernel/file.py

```diff
@@ -96,6 +97,25 @@ def __init__(self,
         super().__init__(path, KernelFileType.KERNEL)
         self.internal_version = self.read_internal_version()
 
+    def decompress_lzma(self, data):
+        results = []
+        while True:
+            decomp = LZMADecompressor(FORMAT_AUTO, None, None)
+            try:
+                res = decomp.decompress(data)
+            except LZMAError:
+                if results:
+                    break  # Leftover data is not a valid LZMA/XZ stream; ignore it.
+                else:
+                    raise  # Error on the first iteration; bail out.
+            results.append(res)
+            data = decomp.unused_data
+            if not data:
+                break
+            if not decomp.eof:
```

Comment:
I don't get this. Does this mean that we expect EOF for all but the final chunk?

Reply:
When I debug it, the `decomp.eof` is always `False`. That's mean that EOF was not properly set when compressing the data (?) . And because `decomp.unused_data` is always an empty byte it will break from the loop before reaching the ` raise LZMAError("Compressed data ended before the end-of-stream marker was reached")` code.

Reply:
Is the loop needed at all then? I suppose the SO solution might have covered a more general case (i.e. multiple concatenated LZMA streams) while the kernel doesn't do that.

Reply:
Yeah I think in my case, the loop is not needed. My solution is based on python's original `lzma.decompress(data, format=FORMAT_AUTO, memlimit=None, filters=None)` function that being used in the eclean-kernel, so that is why the loop is present.

Reply:
Because you mention that the kernel doesn't do multiple concatenated LZMA streams, I try to modify my code and it still works

``` python    
def decompress_lzma(self, data: bytes) -> bytes:
        decomp = LZMADecompressor()
        try:
            res = decomp.decompress(data)
        except LZMAError:
            raise
        return res```

Reply:
> Could you remove it then? Let's keep the code simple, and we can improve it later if it becomes necessary.

Yes I could